### PR TITLE
Remove volume expansion feature gates

### PR DESCRIPTION
**Pull request by mengjiao-liu**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
The volume expansion feature gates were GAed in Kubernetes v1.24 so that we can remove them in 1.27.

- ExpandCSIVolumes
- ExpandInUsePersistentVolumes
- ExpandPersistentVolumes

#### Which issue(s) this PR fixes:

xref:
- https://github.com/kubernetes/enhancements/issues/284
- https://github.com/kubernetes/kubernetes/pull/108929

#### Does this PR introduce a user-facing change?

```release-note
The following feature gates for volume expansion GA features have been removed and must no longer be referenced in `--feature-gates` flags: ExpandCSIVolumes, ExpandInUsePersistentVolumes, ExpandPersistentVolumes
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by mengjiao-liu** [comment]
Wait until the code freeze ends.
/hold

**Comment by dims** [comment]
This PR has the label work-in-progress, please revisit to see if you still need this, please close if not

File: pkg/apis/core/types.go [hunk 1]
```diff
@@ -1741,7 +1741,7 @@ type CSIPersistentVolumeSource struct {
 	// ControllerExpandSecretRef is a reference to the secret object containing
 	// sensitive information to pass to the CSI driver to complete the CSI
 	// ControllerExpandVolume call.
-	// This is an beta field and requires enabling ExpandCSIVolumes feature gate.
+	// This is a beta field.
```
Comment (liggitt) [inline pkg/apis/core/types.go:1744]: Is this still beta or GA? If beta, it should still mention the feature gate that guards it

File: staging/src/k8s.io/api/core/v1/types.go [hunk 2]
```diff
@@ -1826,7 +1826,7 @@ type CSIPersistentVolumeSource struct {
 	// controllerExpandSecretRef is a reference to the secret object containing
 	// sensitive information to pass to the CSI driver to complete the CSI
 	// ControllerExpandVolume call.
-	// This is an beta field and requires enabling ExpandCSIVolumes feature gate.
+	// This is a beta field.
```
Comment (liggitt) [inline staging/src/k8s.io/api/core/v1/types.go:1829]: Same comment about dropping the beta language or keeping a feature gate reference, depending on whether the field is beta or not

File: pkg/apis/core/types.go (hunk 1 above)
Comment (mengjiao-liu) [reply pkg/apis/core/types.go:1744]: I took a look at [the KEP](https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/556-csi-volume-resizing#supporting-per-pvc-secret-refs) and [the PR on which this field was added](https://github.com/kubernetes/kubernetes/commit/69393291b64a2281533cb2567f67edca478b4a93). This field is controlled by the `ExpandCSIVolumes` function gate, so it has reached GA.
```
This API change will be gated by ExpandCSIVolumes feature gate
```

I will remove the content: "This is a beta field."

File: staging/src/k8s.io/api/core/v1/types.go (hunk 2 above)
Comment (mengjiao-liu) [reply staging/src/k8s.io/api/core/v1/types.go:1829]: Same as above. Thanks!

**Comment by liggitt** [comment]
Open a PR to https://github.com/kubernetes/website/ (dev-1.27 branch) updating https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates-removed/ and https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/ pages for these

**Comment by mengjiao-liu** [comment]
> Open a PR to https://github.com/kubernetes/website/ (dev-1.27 branch) updating https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates-removed/ and https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/ pages for these

Ok. Will do the update.

**Comment by mengjiao-liu** [comment]
https://github.com/kubernetes/website/pull/38545 in branch dev-1.27 to document it.

**Pull request merged**